### PR TITLE
fix(j2cl): keep selected-wave version/hash basis atomic

### LINKED ISSUE #936: Keep J2CL selected-wave version/hash basis atomic
## Summary

`J2clSelectedWaveProjector.buildWriteSession(...)` currently advances `baseVersion` and `historyHash` independently. If a selected-wave update carries a newer `resultingVersion` but omits the matching `resultingVersionHistoryHash`, the projector can combine the new version with the previous hash and publish an invalid hashed-version pair for the next reply submit.

## Why this matters

The sidecar write path now depends on the projected write session for reply submits. A mismatched version/hash pair can make the client look ready to reply while constructing a delta basis the server should reject.

## Expected fix

- treat `resultingVersion` and `resultingVersionHistoryHash` as a coupled value
- only advance the write-session basis when both arrive together
- otherwise preserve the previous pair as-is
- add a regression test covering a selected-wave update that changes version without a matching history hash

## Context

Follow-up from merged PR #935.

### PR DESCRIPTION
**Pull request by vega113**
## Summary
- Treat `resultingVersion` and `resultingVersionHistoryHash` as a coupled atomic pair in `J2clSelectedWaveProjector.buildWriteSession`: advance both when the update supplies both, otherwise preserve the previous pair as-is.
- Remove the snapshot/document-version fallback (`resolveBaseVersion`) that silently promoted a version without a matching hash.
- Add `J2clSelectedWaveProjectorTest` with the regression case from the issue plus advance, preserve-on-missing-version, no-previous, and first-coupled-update paths.

## Why
The sidecar reply submit path now depends on the projected write session. Under the old logic, a selected-wave update carrying a newer version but an empty history hash would combine the new version with the previous hash, publishing an invalid hashed-version pair that the server would reject when the client tried to reply.

## Verification
```
./j2cl/mvnw -f j2cl/pom.xml -Psearch-sidecar \
  -Dtest=org.waveprotocol.box.j2cl.search.J2clSelectedWaveProjectorTest,org.waveprotocol.box.j2cl.search.J2clSelectedWaveControllerTest \
  test
```
Result: `Tests run: 20, Failures: 0, Errors: 0, Skipped: 0 — BUILD SUCCESS`.

## Plan & review
- Plan: `docs/superpowers/plans/2026-04-22-issue-936-selected-wave-basis-atomic.md`
- Plan review (Claude Opus 4.7): CLEAN
- Implementation review (Claude Opus 4.7): CLEAN

Closes #936.

## Summary by CodeRabbit

* **Bug Fixes**
  * Treat write-session version and history-hash as an atomic paired basis so partial updates no longer create mismatched version/hash combinations.

* **Tests**
  * Added unit tests covering coupled-update advancement, preservation when history-hash is missing, fallback behavior, and missing-prior-session edge cases.

* **Documentation**
  * Added a changelog entry and a proposed plan documenting the behavioral change and verification steps.

### CODE REVIEW

File: j2cl/src/main/java/org/waveprotocol/box/j2cl/search/J2clSelectedWaveProjector.java [hunk 1]
```diff
@@ -67,22 +67,22 @@ private static J2clSidecarWriteSession buildWriteSession(
...
+      return null;
     }
-    if (channelId == null
-        || channelId.isEmpty()
-        || replyTargetBlipId == null
-        || baseVersion < 0
-        || historyHash == null
-        || historyHash.isEmpty()) {
+    if (channelId == null || channelId.isEmpty() || replyTargetBlipId == null) {
       return null;
```
[the remark this answers is not in the document]
Comment (vega113) [reply j2cl/src/main/java/org/waveprotocol/box/j2cl/search/J2clSelectedWaveProjector.java]: Fixed. Added `|| replyTargetBlipId.isEmpty()` to align the final guard with the earlier fallback logic at lines 65-68.
